src/systems/camera-system.js

```diff
@@ -86,33 +132,79 @@ export class CameraSystem {
 
   inspect(o) {
     if (this.mode !== CAMERA_MODE_INSPECT) {
-      this.preInspectMode = this.mode;
+      this.snapshot.mode = this.mode;
     }
     this.mode = CAMERA_MODE_INSPECT;
     this.inspected = o;
+
+    (getBatch(o, this.batchManagerSystem) || o).traverse(enableInspectLayer);
+
+    const scene = AFRAME.scenes[0];
+    const vrMode = scene.is("vr-mode");
+    const camera = vrMode ? scene.renderer.vr.getCamera(scene.camera) : scene.camera;
+    this.snapshot.mask = camera.layers.mask;
+    camera.layers.set(CAMERA_LAYER_INSPECT);
+    if (vrMode) {
+      this.snapshot.mask0 = camera.cameras[0].layers.mask;
+      this.snapshot.mask1 = camera.cameras[1].layers.mask;
+      camera.cameras[0].layers.set(CAMERA_LAYER_INSPECT);
+      camera.cameras[1].layers.set(CAMERA_LAYER_INSPECT);
+    }
+
     moveRigSoCameraLooksAtObject(this.rigEl.object3D, this.cameraEl.object3D, this.inspected);
+
+    this.snapshot.audio = getAudio(o);
+    if (this.snapshot.audio) {
+      this.snapshot.audio.applyMatrix(new THREE.Matrix4().identity()); // hack around our matrix optimizations
+      this.snapshot.audio.updateMatrices();
+      this.snapshot.audioTransform.copy(this.snapshot.audio.matrixWorld);
+      scene.audioListener.updateMatrices();
+      setMatrixWorld(this.snapshot.audio, scene.audioListener.matrixWorld);
+    }
   }
 
   uninspect() {
-    this.inspected = null;
+    if (this.inspected) {
+      (getBatch(this.inspected, this.batchManagerSystem) || this.inspected).traverse(disableInspectLayer);
+    }
+    const vrMode = AFRAME.scenes[0].is("vr-mode");
+    const scene = AFRAME.scenes[0];
+    const camera = vrMode ? scene.renderer.vr.getCamera(scene.camera) : scene.camera;
+    camera.layers.mask = this.snapshot.mask;
+    if (vrMode) {
+      camera.cameras[0].layers.mask = this.snapshot.mask0;
+      camera.cameras[1].layers.mask = this.snapshot.mask1;
+    }
+    if (this.inspected) {
+      this.inspected = null;
+    }
     if (this.mode !== CAMERA_MODE_INSPECT) return;
-    this.mode = this.preInspectMode || CAMERA_MODE_FIRST_PERSON;
-    this.preInspectMode = null;
+    this.mode = this.snapshot.mode || CAMERA_MODE_FIRST_PERSON;
+    this.snapshot.mode = null;
+
+    if (this.snapshot.audio) {
+      setMatrixWorld(this.snapshot.audio, this.snapshot.audioTransform);
```

Comment:
i suggested we switch to using rolloff in my other comments, but if we stick with this, you should dereference this.snapshot.audio so we don't pin it to memory if the object is removed.

Reply:
:+1: